pkg/infra/kubernetes/exec_unit.go

```diff
@@ -150,6 +156,27 @@ func (unit *HelmExecUnit) transformDeployment(cfg config.ExecutionUnit) ([]HelmC
 
 	deployment.Spec.Template.Spec.ServiceAccountName = unit.getServiceAccountName()
 
+	if deployment.Spec.Template.Spec.NodeSelector == nil {
+		deployment.Spec.Template.Spec.NodeSelector = make(map[string]string)
+	}
+
+	if cfg.NetworkPlacement != "" {
+		deployment.Spec.Template.Spec.NodeSelector["network_placement"] = cfg.NetworkPlacement
+	}
+	if kconfig := cfg.GetExecutionUnitParamsAsKubernetes(); kconfig.InstanceType != "" {
+		instanceTypeKey := fmt.Sprintf("{{ .Values.%sInstanceTypeKey }}", unit.Name)
+		instanceTypeValue := fmt.Sprintf("{{ .Values.%sInstanceTypeValue }}", unit.Name)
+		deployment.Spec.Template.Spec.NodeSelector[instanceTypeKey] = instanceTypeValue
+		values = append(values,
+			HelmChartValue{
+				ExecUnitName: unit.Name,
+				Kind:         deployment.Kind,
+				Type:         string(InstanceTypeKey),
+				Key:          instanceTypeKey,
+			},
+		)
```

Comment:
Yeah, I forgot to go back and add it when I added to EKS as a value. I wasn't sure whether the value could go directly in here like network_placement or not.

Reply:
oh i guess it depends if we have that name at the time of writing this, i know we actually used pulumi output to set it so thats what i was expecting. If it works without that we could do it, but then you are modifying the manifest again rather than setting values on the pulumi helm chart, which seems odd to do outside of the kubernetes plugin

Reply:
Added the value in the latest commit. 👍